projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Invenio angular core
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'ng-core-record-search-page',
+  templateUrl: './record-search-page.component.html',
+  styles: []
+})
+export class RecordSearchComponent implements OnInit {
+  /**
+   * Current filters applied
+   */
+  aggFilters = [];
+
+  /**
+   * Current selected resource type
+   */
+  currentType = 'documents';
+
+  /**
+   * URL to notice detail
+   */
+  detailUrl: string = null;
+
+  /**
+   * Display search input
+   */
+  showSearchInput = true;
+
+  /**
+   * Admin mode (edit, remove, add, ...)
+   */
+  adminMode = true;
+
+  /**
+   * Define the current record's page
+   */
+  page = 1;
+
+  /**
+   * Define the number of records per page
+   */
+  size = 10;
+
+  /**
+   * Search query
+   */
+  q = '';
+
+  /**
+   * Types of resources available
+   */
+  types: {
+    key: string,
+    label: string,
+    component?: Component,
+    total?: number,
+    canAdd?: any,
+    canUpdate?: any,
+    canDelete?: any,
+    canRead?: any,
+    aggregations?: any
+  }[] = [{ key: 'documents', label: 'Documents' }];
+
+  /**
+   * Constructor
+   * @param dialogService - Modal component
+   * @param recordService - Service for managing records
+   * @param toastService - Toast message
+   * @param route - Angular current route
+   * @param router - Angular router
+   */
+  constructor(
+    private route: ActivatedRoute,
+    protected router: Router
+  ) { }
+
+  /**
+   * Component initialisation.
+   */
+  ngOnInit() {
+    const queryParams = this.route.snapshot.queryParams;
+    for (const key in queryParams) {
+      if (['q', 'page', 'size'].includes(key)) {
+        this[key] = queryParams[key];
+      } else {
+        if (Array.isArray(queryParams[key]) === false) {
```

Comment:
if not?

Reply:
I don't understand this comment, there's a "else" with this condition.

Reply:
I propose here:
```if (!Array.isArray(queryParams[key]))```